redis/client.py

```diff
@@ -1411,6 +1432,20 @@ def get_message(self, ignore_subscribe_messages=False, timeout=0):
         before returning. Timeout should be specified as a floating point
         number.
         """
+        if not self.subscribed:
+            # Wait for subscription
+            start_time = time.time()
+            if self.subscribed_event.wait(timeout) is True:
+                # The connection was subscribed during the timeout frametime.
+                # The timeout should be adjusted based on the time spent
+                # waiting for the subscription
+                time_spent = time.time() - start_time
+                timeout = timeout - time_spent
```

Comment:
parse_response raises an error when given a negative timeout, which could occur if the event is set just as the timeout expires.
```suggestion
                timeout = max(0.0, timeout - time_spent)
```
Possibly this function should also raise if the initial timeout is negative, so that a negative timeout from the user has consistent behaviour regardless of whether there are subscriptions.

Reply:
changed